lib/matplotlib/axes/_axes.py

```diff
@@ -794,6 +794,69 @@ def axvline(self, x=0, ymin=0, ymax=1, **kwargs):
         return l
 
     @docstring.dedent_interpd
+    def axline(self, slope=1, intercept=0, **kwargs):
+        """
+        Add an infinite line across the axes.
+
+        Parameters
+        ----------
+        slope : scalar, optional, default: 1
+            Slope of the line.
+
+        intercept : scalar, optional, default: 0
+            Intercept of the line with the y-axis.
+
+        Returns
+        -------
+        :class:`~matplotlib.lines.Line2D`
+
+        Other Parameters
+        ----------------
+        Valid kwargs are :class:`~matplotlib.lines.Line2D` properties,
+        with the exception of 'transform':
+
+        %(Line2D)s
+
+        Examples
+        --------
+        * draw a thick red line with slope 1 and y-intercept 0::
+
+            >>> axline(linewidth=4, color='r')
+
+        * draw a default line with slope 1 and y-intercept 1::
+
+            >>> axline(intercept=1)
+
+        * draw a default line with slope 5 and y-intercept 0::
+
+            >>> axline(slope=5)
+
+        See Also
+        --------
+        axhline : for strictly horizontal lines
+        axvline : for strictly vertical lines
```

Comment:
Do these get automatically linked? If not would be nice to have some sphinx markup to link to relevant methods.